test_files/test.rb

```diff
@@ -0,0 +1,46 @@
+require "net/http"
+
+class SomeClass
+  def nice()
+    "Nice"
+  end
+end
+
+module Foo
+  def foo
+    :foo
+  end
+end
+
+module Test::SubTest
+  class MyClass < SomeClass
+    include Foo
+
+    def self.foo
```

Comment:
No - and I can't work it out. So given this:

```ruby
    def self.in_eigenclass
      new.simple
    end
```
Here's what `./parse -file test_files/test.rb -tags_query_dir queries/` is spitting out:

```
          singleton_method [27, 4] - [29, 7] "def self.in_eigenclass\n…"
            "def" [27, 4] - [27, 7] "def"
            object: self [27, 8] - [27, 12] "self"
            "." [27, 12] - [27, 13] "."
            name: identifier [27, 13] - [27, 26] "in_eigenclass"
            body: body_statement [28, 6] - [28, 16] "new.simple"
```

and I would expect that to be caught by

```
    (singleton_method
      name: (_) @name
      parameters: (method_parameters) @codeium.parameters
      body: (_) @body
    )
```
but it isn't, and I have no idea why.

Similarly, the method definition inside `class << self` is appearing as a `method` according to the parser (which I suspect the parser can't do anything about, despite the code in my test being functionally equivalent to singleton_method), yet despite it being otherwise a "normal" method according to the parser, it isn't being picked up by the query.

Even more confusing, the method `splatted_args` isn't being picked up. The `*args` is similar to `...args` in javascript - yet `...args` works fine when added to test.js but the ruby query is ignoring it.

Unfortunately, I'm beyond the limit of my abilities with the tree-sitter syntax - any ideas of things we could try?

Reply:
Ah, I think I know what's happening. You'll need to mark the parameters node as optional, similar to this, for example: https://github.com/Exafunction/codeium-parse/blob/7be8f1c9166acf0c9d544393ed895468e9999e29/queries/java_tags.scm#L20

Reply:
That's now done, and that has seemed to work, though I'm still confused how! (There were other params-less methods getting caught fine)  :+1: